workspaces/global-floating-action-button/plugins/global-floating-action-button/src/types.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export enum Position {
+  bottomRight = 'bottomRight',
+  bottomCenter = 'bottomCenter',
+}
+
+export type FloatingActionButtonWithPositions = Record<
+  Position | string,
+  FloatingActionButton[]
+>;
+
+export type FloatingActionButton = {
+  label: string;
+  showLabel?: boolean;
+  icon: React.ReactElement;
```

Comment:
As mentioned above, we should allow a string here as well. Just here again to see that it's also resolved here. I think we can make it optional? There should be at least a label then. So render the label when `showLabel || !icon` ;-)

```suggestion
  icon?: string | React.ReactElement;
```

Reply:
@christoph-jerolimov  I was under the impression that the  icon is mandatory, if the user wants to show a label they can do so and that will appear next to the icon

Reply:
I think less is more and in this case this means: It's easier for an admin to a (temporary) FAB without an icon then thinking on "where I get this icon from".